### PR TITLE
[PATCH v3] Master pause enq test

### PR DESCRIPTION
**Pull request by psavol**
Add scheduler test case to check that queues enqueues work with pause on.

### CODE REVIEW

**Review by shijith — COMMENTED** [review]
Tested on CN10K.
Tested-by: Shijith Thotton <sthotton@marvell.com>

File: test/validation/api/scheduler/scheduler.c [hunk 1]
```diff
@@ -1847,6 +1851,109 @@ static void scheduler_test_pause_resume(void)
...
+
+	for (i = 0; i < NUM_BUFS_BEFORE_PAUSE; i++) {
+		ev = ev_tbl[i];
+		ret = odp_queue_enq(queue, ev);
+		CU_ASSERT_FATAL(ret == 0);
+	}
+
+	local_bufs = 0;
+	while (1) {
+		ev = odp_schedule(&from, ODP_SCHED_NO_WAIT);
```
Comment (matiaselo) [inline test/validation/api/scheduler/scheduler.c:1899]: `from` is not initialized to `ODP_QUEUE_INVALID`

File: test/validation/api/scheduler/scheduler.c [hunk 2]
```diff
@@ -1847,6 +1851,109 @@ static void scheduler_test_pause_resume(void)
...
+		CU_ASSERT(from == queue);
+		ev_tbl[i] = ev;
+	}
+
+	/* Pause, schedule, enqueue, resume */
+	odp_schedule_pause();
+
+	local_bufs = 0;
+	while (1) {
+		ev = odp_schedule(&from, ODP_SCHED_NO_WAIT);
```
Comment (matiaselo) [inline test/validation/api/scheduler/scheduler.c:1926]: `from` is not initialized to `ODP_QUEUE_INVALID`

File: test/validation/api/scheduler/scheduler.c [hunk 3]
```diff
@@ -1847,6 +1851,109 @@ static void scheduler_test_pause_resume(void)
...
+		ev = ev_tbl[i];
+		ret = odp_queue_enq(queue, ev);
+		CU_ASSERT_FATAL(ret == 0);
+	}
+
+	odp_schedule_resume();
+
+	/* Free all */
+	for (i = 0; i < NUM_BUFS_PAUSE; i++) {
+		ev = odp_schedule(&from, ODP_SCHED_WAIT);
```
Comment (matiaselo) [inline test/validation/api/scheduler/scheduler.c:1948]: May deadlock if events are lost. Perhaps simply call `drain_queues()` and check the return value.

File: test/validation/api/scheduler/scheduler.c [hunk 4]
```diff
@@ -173,10 +177,11 @@ static void scheduler_test_wait_time(void)
 	queue = odp_queue_create("dummy_queue", &qp);
 	CU_ASSERT_FATAL(queue != ODP_QUEUE_INVALID);
 
-	wait_time = odp_schedule_wait_time(ODP_TIME_SEC_IN_NS);
+	wait_time = odp_schedule_wait_time(WAIT_TIMEOUT);
 	start_time = odp_time_local();
-	odp_schedule(&queue, ODP_SCHED_NO_WAIT);
+	ev = odp_schedule(NULL, ODP_SCHED_NO_WAIT);
 	end_time = odp_time_local();
+	CU_ASSERT(ev == ODP_EVENT_INVALID);
```
Comment (matiaselo) [inline test/validation/api/scheduler/scheduler.c:184]: Possible event is not freed.

File: test/validation/api/scheduler/scheduler.c (hunk 1 above)
Comment (psavol) [reply test/validation/api/scheduler/scheduler.c:1899]: fixed in v3

File: test/validation/api/scheduler/scheduler.c (hunk 2 above)
Comment (psavol) [reply test/validation/api/scheduler/scheduler.c:1926]: fixed in v3

File: test/validation/api/scheduler/scheduler.c (hunk 3 above)
Comment (psavol) [reply test/validation/api/scheduler/scheduler.c:1948]: fixed in v3

File: test/validation/api/scheduler/scheduler.c (hunk 4 above)
Comment (psavol) [reply test/validation/api/scheduler/scheduler.c:184]: Fixed in v3. Changed to CU_ASSERT_FATAL as no event is expected.

**Review by matiaselo — APPROVED** [review]

**Pull request merged**